File: lib-database-entities/src/main/java/com/ustadmobile/lib/db/entities/Schedule.java
package com.ustadmobile.lib.db.entities;

import com.ustadmobile.lib.database.annotation.UmEntity;
import com.ustadmobile.lib.database.annotation.UmPrimaryKey;

@UmEntity
public class Schedule implements SyncableEntity {

    public static final int SCHEDULE_FREQUENCY_DAILY = 1;
    public static final int SCHEDULE_FREQUENCY_WEEKLY = 2;

    public static final int SCHEDULE_FREQUENCY_ONCE = 3;
    public static final int SCHEDULE_FREQUENCY_MONTHLY = 4;
    public static final int SCHEDULE_FREQUENCY_YEARLY = 5;

    public static final int DAY_SUNDAY = 1;
    public static final int DAY_MONDAY = 2;
    public static final int DAY_TUESDAY = 3;
    public static final int DAY_WEDNESDAY = 4;
    public static final int DAY_THURSDAY = 5;
    public static final int DAY_FRIDAY = 6;
    public static final int DAY_SATURDAY = 7;


    public static final int MONTH_JANUARY = 1;
    public static final int MONTH_FEBUARY = 2;
    public static final int MONTH_MARCH = 3;
    public static final int MONTH_APRIL = 4;
    public static final int MONTH_MAY = 5;
    public static final int MONTH_JUNE = 6;
    public static final int MONTH_JULY = 7;
    public static final int MONTH_AUGUST = 8;
    public static final int MONTH_SEPTEMBER = 9;
    public static final int MONTH_OCTOBER = 10;
    public static final int MONTH_NOVEMBER = 11;
    public static final int MONTH_DECEMBER = 12;


    @UmPrimaryKey(autoIncrement = true)
    private long scheduleUid;

    //Start time
    private long sceduleStartTime;

    //End time
    private long scheduleEndTime;

    //What day for this frequency
    private int scheduleDay;

    //What month for this frequency
    private int scheduleMonth;

    // Frequency - Once, Daily, Every Week, Every Month, Every Year
    private int scheduleFrequency;

    //The Calendar this will be set to.
    private long umCalendarUid;

    //What clazz is this Schedule for
    private long scheduleClazzUid;

    //active or removed
    private boolean scheduleActive;

    public boolean isScheduleActive() {
        return scheduleActive;
    }

    public void setScheduleActive(boolean scheduleActive) {
        this.scheduleActive = scheduleActive;
    }

    public long getScheduleClazzUid() {
        return scheduleClazzUid;
    }

    public void setScheduleClazzUid(long scheduleClazzUid) {
        this.scheduleClazzUid = scheduleClazzUid;
    }

    public long getSceduleStartTime() {
        return sceduleStartTime;
    }

    public void setSceduleStartTime(long sceduleStartTime) {
        this.sceduleStartTime = sceduleStartTime;
    }

    public long getScheduleEndTime() {
        return scheduleEndTime;
    }

    public void setScheduleEndTime(long scheduleEndTime) {
        this.scheduleEndTime = scheduleEndTime;
    }

    public int getScheduleDay() {
        return scheduleDay;
    }

    public void setScheduleDay(int scheduleDay) {
        this.scheduleDay = scheduleDay;
    }

    public int getScheduleMonth() {
        return scheduleMonth;
    }

    public void setScheduleMonth(int scheduleMonth) {
        this.scheduleMonth = scheduleMonth;
    }


    public int getScheduleFrequency() {
        return scheduleFrequency;
    }

    public void setScheduleFrequency(int scheduleFrequency) {
        this.scheduleFrequency = scheduleFrequency;
    }

    public long getUmCalendarUid() {
        return umCalendarUid;
    }

    public void setUmCalendarUid(long umCalendarUid) {
        this.umCalendarUid = umCalendarUid;
    }

    public long getScheduleUid() {
        return scheduleUid;
    }

    public void setScheduleUid(long scheduleUid) {
        this.scheduleUid = scheduleUid;
    }

    @Override
    public long getMasterChangeSeqNum() {
        return 0;
    }

    @Override
    public void setMasterChangeSeqNum(long masterChangeSeqNum) {

    }

    @Override
    public long getLocalChangeSeqNum() {
        return 0;
    }

    @Override
    public void setLocalChangeSeqNum(long localChangeSeqNum) {

    }
}

File: lib-database-entities/src/commonMain/kotlin/com/ustadmobile/staging/lib/db/entities/Schedule.kt
package com.ustadmobile.lib.db.entities

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.ustadmobile.door.annotation.LastChangedBy
import com.ustadmobile.door.annotation.LocalChangeSeqNum
import com.ustadmobile.door.annotation.MasterChangeSeqNum
import com.ustadmobile.door.annotation.SyncableEntity
import kotlinx.serialization.Serializable

@SyncableEntity(tableId = 21)
@Entity
@Serializable
class Schedule {

    @PrimaryKey(autoGenerate = true)
    var scheduleUid: Long = 0

    //Start time
    /**
     * Get the time of day that this schedule is to begin. This should be in ms from the beginning of
     * the day. E.g. 14:30 = (14.5 * 60 * 60 * 1000) ms
     *
     * @return time of the day that class is to begin for this scheduled instance
     */
    /**
     * Set the time of day that this schedule is to begin. This should be in ms from the beginning of
     * the day. E.g. 14:30 = (14.5 * 60 * 60 * 1000) ms
     *
     * @param sceduleStartTime time of the day that class is to begin for this scheduled instance
     */
    var sceduleStartTime: Long = 0

    //End time
    /**
     *
     * @return
     */
    var scheduleEndTime: Long = 0

    //What day for this frequency
    var scheduleDay: Int = 0

    //What month for this frequency
    var scheduleMonth: Int = 0

    // Frequency - Once, Daily, Every Week, Every Month, Every Year
    var scheduleFrequency: Int = 0

    //The Calendar this will be set to.
    var umCalendarUid: Long = 0

    //What clazz is this Schedule for
    var scheduleClazzUid: Long = 0

    @MasterChangeSeqNum
    var scheduleMasterChangeSeqNum: Long = 0

    @LocalChangeSeqNum
    var scheduleLocalChangeSeqNum: Long = 0

    @LastChangedBy
    var scheduleLastChangedBy: Int = 0

    //active or removed
    var scheduleActive: Boolean = false

    companion object {

        val SCHEDULE_FREQUENCY_DAILY = 1
        val SCHEDULE_FREQUENCY_WEEKLY = 2

        val SCHEDULE_FREQUENCY_ONCE = 3
        val SCHEDULE_FREQUENCY_MONTHLY = 4
        val SCHEDULE_FREQUENCY_YEARLY = 5

        val DAY_SUNDAY = 1
        val DAY_MONDAY = 2
        val DAY_TUESDAY = 3
        val DAY_WEDNESDAY = 4
        val DAY_THURSDAY = 5
        val DAY_FRIDAY = 6
        val DAY_SATURDAY = 7


        val MONTH_JANUARY = 1
        val MONTH_FEBUARY = 2
        val MONTH_MARCH = 3
        val MONTH_APRIL = 4
        val MONTH_MAY = 5
        val MONTH_JUNE = 6
        val MONTH_JULY = 7
        val MONTH_AUGUST = 8
        val MONTH_SEPTEMBER = 9
        val MONTH_OCTOBER = 10
        val MONTH_NOVEMBER = 11
        val MONTH_DECEMBER = 12
    }
}
